src/lib.rs

```diff
@@ -377,9 +377,9 @@ impl Receiver {
 
             try_handle_http(&server, &http_histogram, &gauges, &global_counters);
 
-            let t1 = time::precise_time_ns();
+            let t1 = Instant::now();
 
-            if t1 - t0 >= (duration as u64 * ONE_SECOND) {
+            if (t1 - t0).as_secs() >= (duration as u64 * ONE_SECOND) {
```

Comment:
One of this issues is here, this should be changed to: (as duration is already in seconds, this was seconds -> ns conversion)
```rust
if (t1 - t0).as_secs() >= (duration as u64) {
```